internal/wasm/call_context.go

```diff
@@ -163,15 +149,12 @@ func (f *FunctionInstance) ResultTypes() []api.ValueType {
 }
 
 // Call implements the same method as documented on api.Function
-func (f *FunctionInstance) Call(m api.Module, params ...uint64) (ret []uint64, err error) {
-	mod := f.Module
-	modCtx, ok := m.(*CallContext)
-	if ok {
-		// TODO: check if the importing context is correct
```

Comment:
cool thing is we no longer have this TODO! before, we allowed the user (by accident) to change the module input. We did that only to allow the context to get through. Now that we separated the types, we no longer have that bomb.